src/main/java/juja/microservices/gamification/controller/AchievementController.java

```diff
@@ -34,4 +35,12 @@
         List<String> ids = achievementService.addThanks(request.getFrom(), request.getTo(), request.getDescription());
         return ResponseEntity.ok(ids);
     }
+
+    @RequestMapping(value = "/achieve/codenjoy", method = RequestMethod.POST)
+    @ResponseBody
+    public ResponseEntity<?> addCodenjoy(@RequestBody CodenjoyRequest request) {
+        List<String> ids = achievementService.addCodenjoy(request.getFrom(), request.getFirstPlace(),
```

Comment:
I think you can pass to service the whole object CodenjoyRequest and get fuilds from this object there - on the service layer.